### PR TITLE
Fix the KeybindingResetDetector

### PR DESCRIPTION
**Pull request by jasonmalinowski**
We switched VisualStudioInfoBar to no longer be a MEF service in 347cd02a80446b47eb93d1effc6d675810457bc2, but this was still importing it. It's now a helper type, so create it instead.

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]

**Review by jmarolf — APPROVED** [review]

File: src/VisualStudio/Core/Def/KeybindingReset/KeybindingResetDetector.cs [hunk 1]
```diff
@@ -91,13 +92,13 @@ internal sealed class KeybindingResetDetector : ForegroundThreadAffinitizedObjec
         public KeybindingResetDetector(
             IThreadingContext threadingContext,
             IGlobalOptionService globalOptions,
-            VisualStudioInfoBar infoBarService,
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/KeybindingReset/KeybindingResetDetector.cs:94]: i don't get it.  shouldn't this have broken mef?  how was this type getting instantiated?  and how did this not break anything else importing KeybindingResetDetector?

**Review by CyrusNajmabadi — CHANGES_REQUESTED** [review]
Something doesn't make sense to me here.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi: It throws exceptions here:

https://github.com/dotnet/roslyn/blob/91bd6e559166e4550e4e975d827ef301b4d62d1e/src/VisualStudio/Core/Def/RoslynPackage.cs#L260-L261

Which is a fire-and-forget path.

**Comment by CyrusNajmabadi** [comment]
bleaghhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi Updated to report non-fatal telemetry if that throws.

**Comment by jasonmalinowski** [comment]
Some testing makes us think this feature is more generally broken; until we understand why we'll merge this for 17.3 to avoid the MEF composition issue, and we'll need to investigate further.

**Pull request merged**